lib/libutee/include/tee_api_defines.h

```diff
@@ -137,6 +137,7 @@
 #define TEE_ALG_DES3_CBC_MAC_PKCS5              0x30000513
 #define TEE_ALG_SM4_ECB_NOPAD                   0x10000014
 #define TEE_ALG_SM4_CBC_NOPAD                   0x10000114
+#define TEE_ALG_SM4_XTS                         0x10000414
```

Comment:
From which specification?

Reply:
Thanks for your reply. The 0x10000414 is assembled according to the assembly rules. Bits[31:28] are TEE_OPERATION_CIPHER, Bits[11:8] are TEE_CHAIN_MODE_XTS, and Bits[7:0] are TEE_MAIN_ALGO_SM4.

Reply:
Yes, that's the logical choice. But it would also be nice to know if this is explicitly defined too since this way of constructing an ID was removed in version 1.2 of the spec.

I can't find this in any public spec, but it would be good to know if you know that it's part of a draft or otherwise planned.

Reply:
Yes. The spec doesn't have this algorithm, but we really need to use it. SM4-XTS is widely used and our customers have the same request. If I want to add this algorithm, how do I define it?

Reply:
We should be able to find a way to accommodate this. @jforissier, what's your take on this?

Reply:
`TEE_ALG_SM4_XTS` should be added to `tee_api_defines_extensions.h` instead of `tee_api_defines.h`, like was done for  [`TEE_ALG_DES3_CMAC`](https://github.com/OP-TEE/optee_os/commit/eee637e7c) for example. And the value should be in the implementation-specific range: `0xF0000414` would be fine I think.